lib/v2/smashingmagazine/category.js

```diff
@@ -0,0 +1,57 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const { category } = ctx.params;
+    const baseUrl = 'https://www.smashingmagazine.com';
+    let route;
+    if (category) {
+        route = `/category/${category}`;
+    } else {
+        route = '/articles';
+    }
+    const { data: response } = await got(`${baseUrl}${route}`);
+    const $ = cheerio.load(response);
+
+    const listItems = $('article.article--post')
+        .toArray()
+        .map((item) => {
+            item = $(item);
+            const a = item.find('h2.article--post__title a');
+            const description = item.find('p.article--post__teaser').clone().children().remove().end().text();
+            const author = item.find('span.article--post__author-name a').text();
+            const time = $('p.article--post__teaser time').attr('datetime');
+            const pubDate = parseDate(time, 'YYYY-MM-DD');
+            return {
+                title: a.text(),
+                link: `${baseUrl}${a.attr('href')}`,
+                pubDate,
+                description,
+                author,
+            };
+        });
+
+    const items = await Promise.all(
+        listItems.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: response } = await got(item.link);
+                const $ = cheerio.load(response);
+                item.category = $('li.meta-box--tags a')
+                    .toArray()
+                    .map((item) => $(item).text());
+                return item;
```

Comment:
Since you're visiting the full article, you can put the article html into `description` as well.